File: web/src/types/task.ts
export interface Task {
  id: number
  project_id: number
  type: string
  status: 'pending' | 'running' | 'done' | 'failed' | 'cancelled'
  progress: number
  step: number
  step_label: string
  error: string | null
  logs?: string[]
}

export const PARSE_STEP_LABELS = [
  '图像预处理与结构增强',
  'VLM 识别房间列表',
  'VLM 分批提取轮廓',
  'OpenCV 评估墙线质量',
  '生成草稿并质检',
]

File: server/app/schemas/task.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field

from app.models.task import TaskStatus, TaskType


class TaskRead(BaseModel):
    id: int
    project_id: int
    type: TaskType
    status: TaskStatus
    progress: float
    step: int
    step_label: str
    retry_count: int
    payload: str | None
    error: str | None
    logs: list[str] = Field(default_factory=list)
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
